src/sentry/seer/code_review/webhooks.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Mapping
+from typing import Any
+
+import orjson
+from django.conf import settings
+from pydantic import ValidationError
+from urllib3.exceptions import MaxRetryError, TimeoutError
+
+from sentry import options
+from sentry.models.organization import Organization
+from sentry.net.http import connection_from_url
+from sentry.seer.signed_seer_api import make_signed_seer_api_request
+from sentry.utils import metrics
+from sentry.utils.seer import can_use_prevent_ai_features
+
+from .types import GitHubCheckRunAction, GitHubCheckRunEvent
+
+logger = logging.getLogger(__name__)
+
+
+# This needs to match the value defined in the Seer API:
+# https://github.com/getsentry/seer/blob/main/src/seer/routes/codegen.py
+SEER_CODEGEN_PATH = "/v1/automation/codegen"
+SEER_PR_REVIEW_RERUN_PATH = f"{SEER_CODEGEN_PATH}/rerun"
+PREFIX = "seer.code_review.check_run.rerun"
+
+
+def handle_github_check_run_event(organization: Organization, event: Mapping[str, Any]) -> bool:
+    """
+    Handle GitHub check_run webhook events for PR review rerun.
+
+    This is called when a check_run event is received from GitHub.
+    When a user clicks "Re-run" on a check run in GitHub UI, we forward
+    the original run ID to Seer so it can rerun the PR review.
+
+    Args:
+        organization: The Sentry organization
+        event: The webhook event payload
+
+    Returns:
+        True if the event was handled successfully, False otherwise
+    """
+    # Validate event payload using Pydantic
+    try:
+        validated_event = _validate_github_check_run_event(event)
+    except (ValidationError, ValueError):
+        # Handle validation errors to prevent sending a 500 error to GitHub
+        # which would trigger a retry. Both ValidationError (Pydantic) and
+        # ValueError (numeric check) are caught here.
+        metrics.incr(f"{PREFIX}.outcome", tags={"status": "invalid_payload"})
+        return False
+
+    if not _should_handle_github_check_run_event(organization, validated_event.action):
+        return False
+
+    extra: dict[str, Any] = {
+        "organization_id": organization.id,
+        "action": validated_event.action,
+        "html_url": validated_event.check_run.html_url,
+        "external_id": validated_event.check_run.external_id,
```

Comment:
This value is set by Seer when the check is originally created.

See here:
https://github.com/getsentry/seer/blob/0f1291a7bc62afc03dd4b50524956356659744d8/src/seer/automation/codegen/pr_review_step.py#L475-L480